content/blog/2020-08-14-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,242 @@
+Note that in the code chunk below we are reading this file from a folder called `data` so we recommend you create such a folder in your working directory and place the file you downloaded there. 
+You can also follow along with the analysis on RStudio Cloud [here](https://rstudio.cloud/project/1523407).
+
+```{r,message=FALSE}
+library(tidyverse)
+library(readxl)
+
+spending <- read_excel("data/spending.xlsx")
+spending
+```
+
+We take note of the month column as well as amount spent, budget, and, expense type. We'll hold off on looking at the column `num_of_expenses`, it may prove useful down the line but not pressing for a quick look into the data.
+
+
+## Nested logic
+
+One advantage of spreadsheets is that we can use formulas to dictate what should happen in a specific cell and, in a broader application of this, apply them to an entire column.  When creating a column that requires multiple or *nested* logical conditions, we can take advantage of the `case_when()` function from dplyr.  Similar to a CASE statement in SQL, we can set parameters for what a given value should be based on conditions from other rows or columns in our data.
+
+Back to our spending sheet, while data for all 12 months is useful for our summary we want to take a look at spending by quarter, something not provided in the initial data set.  
+To do this we can create a new column using the `mutate()` function and apply `case_when()` within that call. 
+
+```{r}
+spending_quarter <- spending %>%
+  mutate(quarter = case_when(
+    month %in% c("Jan", "Feb", "Mar") ~ "Q1",
+    month %in% c("Apr", "May", "Jun") ~ "Q2",
+    month %in% c("Jul", "Aug", "Sep") ~ "Q3",
+    month %in% c("Oct", "Nov", "Dec") ~ "Q4",
+    TRUE ~ "NA"
+  ))
+
+spending_quarter
+```
+
+Excellent! We've got our new quarter column set up. Here we specified which months make up each quarter and for good measure added an NA condition in the event we've missed something. We can do a quick sanity check for our `case_when()` statement by examining our data and seeing if any NA values show up in the new column we generated, which would indicate if a month is spelled differently than what's listed in our conditions or if there's an error in our code. 
+
+For a direct comparison, we can see how this process works in spreadsheets, manipulating our data in an Excel workbook.
+
+![Nested logic](screenshots/nested-logic.png)
+
+It's a bit difficult to see the formula bar in the image above, so the formula to create used for creating the quarter column is provided below.
+
+```{}
+=IF(OR(A2="Jan",A2="Feb",A2="Mar"),"Q1",IF(OR(A2="Apr",A2="May",A2="Jun"),"Q2",IF(OR(A2="Jul",A2="Aug",A2="Sep"),"Q3",IF(OR(A2="Oct",A2="Nov",A2="Dec"),"Q4","NA"))))
+```
+
+
+## Pivot to this section
```

Comment:
Same comment re: inverting the order here: Excel, then R.

I think that also sets up a nice comparison about how the pivot table equivalent in R is amenable to downstream work, whereas the pivot table in Excel is conceived as a finished product for human eyeballs.

Reply:
I agree and the order like that especially in this section makes sense